hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/FrameBuilder.java

```diff
@@ -204,4 +208,15 @@ private boolean contractMustBePresent(
         final var possiblyGrandFatheredEntityNumOf = contractID.hasContractNum() ? contractID.contractNum() : null;
         return !featureFlags.isAllowCallsToNonContractAccountsEnabled(config, possiblyGrandFatheredEntityNumOf);
     }
+
+    private boolean emptyCodePossiblyAllowed(
+            @NonNull final Configuration config,
+            @NonNull final FeatureFlags featureFlags,
+            @NonNull final ContractID contractId,
+            @NonNull final HederaEvmTransaction transaction,
+            @NonNull final Code code) {
+        return !(contractMustBePresent(config, featureFlags, contractId) && code.equals(CodeV0.EMPTY_CODE))
```

Comment:
⚠️ How we are going to keep complexity under control with compound conditionals like this being sprinkled in? 😵‍💫 

Shouldn't we have already enforced the contract existence check somewhere upstream in `TransactionProcessor#computeInvolvedParties()` (in any case, there is lots of code there related to contract-must-exist)?

Reply:
I must admit that I'm responsible for this monstrosity as I needed this check in order to maintain equivalency with mono service as @stoqnkpL says.  But this code doesn't seem to fit well with `TransactionProcessor#computeInvolvedParties` either.  Maybe it'll be better to create an additional step like`TransactionProcessor#validateContract` after we compute the parties.  That way we can isolate all of the logic for whether we care about contract/account/contract code existence there as private methods or static methods in a utility class?